packages/roosterjs-editor-dom/lib/list/VListItem.ts

```diff
@@ -360,13 +395,28 @@ function createListElement(
         result = doc.createElement(listType == ListType.Ordered ? 'ol' : 'ul');
     }
 
+    // Always maintain the metadata saved in the list
+    if (originalRoot && nextLevel == 1 && listType != getListTypeFromNode(originalRoot)) {
+        const style = getMetadata<ListStyleMetadata>(originalRoot, ListStyleDefinitionMetadata);
+        if (style) {
+            setMetadata(result, style, ListStyleDefinitionMetadata);
```

Comment:
I'm confused here.
So you call getMetadata(), then without any change to the style object you call setMetadata() to save it back. Why we need to do this? It seems nothing is changed at all.

Reply:
The list can be reused a new one is created, so the metadata is lost, to prevent the lost of the metadata, get the metadata from the original root and pass it to the new one.

Reply:
I see, got it.